### DISCUSSION TITLE
Configuration options?

### DISCUSSION THREAD

**Discussion by clphillips** [thephpleague/commonmark#662] [Q&A] [answered]
I'm looking for the configuration options that can be passed to League\CommonMark\CommonMarkConverter::__construct and Environment, but can't find that anywhere in the docs, source docs, or tests.

Am I missing something?

The only other reference I can find is #42, but I can't believe that's an exhaustive list.

**Comment by colinodell** [comment]
There are currently only three configuration options, and they're all under the `renderer` key:

``` php
$defaultConfig = [
    'renderer' => [
        'block_separator' => "\n",
        'inner_separator' => "\n",
        'soft_break'      => "\n",
    ],
];
```

There simply hasn't been a need to configure additional settings, but the mechanism is there should we need it.  #42 demonstrates some examples of how we could expand the configuration in the future.  In addition, it's entirely possible for custom extensions to define their own configuration options as needed.

I'm definitely open to any ideas or PRs you have for new configuration options.

I'll keep this issue open until the current options have been documented properly.

**Comment by clphillips** [comment]
@colinodell, so I'd need to roll my own parser to get an HTML safe version (one that escapes all HTML input, e.g. has no allowed tags)?

**Comment by colinodell** [comment]
Unfortunately that is not currently configurable.  If you want to escape all HTML output, I'd think that excluding the HTML parsers (`HtmlBlockParser` and `RawHtmlParser`) from your environment would work.  (I wonder what would happen if you only excluded one and not the other - that might be something to play with.)

I think having a configuration option for HTML is a fantastic idea though, and it's definitely something I'd want to have implemented before a v1.0 release.

**Comment by colinodell** [comment]
You could also look at replacing the renderers instead (with ones that escape the raw HTML output).

**Comment by clphillips** [comment]
Cool, I'll take a look.

**Comment by colinodell** [comment, marked as answer]
Configuration documentation can now be found at https://commonmark.thephpleague.com/configuration/
